apps/api/src/domain/games/memory-match/queries/memory-match-queries.fetch-many.e2e.spec.ts

```diff
@@ -50,11 +59,65 @@ const unpublishedRound = buildTestInstance(MemoryMatchRound, {
 describe(`when querying for a memory match round: fetch many`, () => {
     let app: INestApplication;
 
-    let databaseProvider: ArangoDatabaseProvider;
-
     let memoryMatchRepository: IMemoryMatchRepository;
 
+    const setItUp = async (testUserWithGroups: CoscradUserWithGroups) => {
+        const mockConfigService = buildMockConfigService(
+            {
+                // TODO can we fix this?
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                GLOBAL_PREFIX: '',
+            },
+            buildConfigFilePath(Environment.test)
+        );
+
+        const testModuleRef = await Test.createTestingModule({
+            imports: [
+                PersistenceModule.forRootAsync(),
+                ConfigModule.forRoot({
+                    isGlobal: true,
+                    envFilePath: buildConfigFilePath(process.env.NODE_ENV),
+                    cache: false,
+                }),
+                AuthorizationModule,
+                MemoryMatchModule,
+            ],
+        })
+            .overrideGuard(OptionalJwtAuthGuard)
+            .useValue(new MockJwtAuthGuard(testUserWithGroups, true))
+            .overrideGuard(AdminJwtGuard)
+            .useValue(new MockJwtAdminAuthGuard(testUserWithGroups))
+            .overrideProvider(ConfigService)
+            .useValue(mockConfigService)
+            .compile();
+
+        app = testModuleRef.createNestApplication();
+
+        await app.init();
+
+        memoryMatchRepository = app.get(MEMORY_MATCH_REPOSITORY_INJECTION_TOKEN);
+    };
+
+    const dummyUser = getValidAggregateInstanceForTest(AggregateType.user).clone({
```

Comment:
Use `buildTestInstance` to build the relevant user at the top of the `describe` block for that given user's test cases. We want to see the user's state very close to the tests without scrolling to the top.